### ISSUE TITLE
A .pdf file gets marked as completed before showing the completion modal to the learner

### ISSUE THREAD

**Issue by pcenov** [learningequality/kolibri#9884]
## Observed behavior
This one was hard to spot in testing, but when one is scrolling down slowly while reading a .pdf file, the completion modal get's displayed after the start icon for the completion is displayed and only once the user has scrolled down completely.

## Expected behavior
The completion modal should be displayed prior or simultaneously with the completion icon.

## Steps to reproduce the issue
1. Install the the following [build ](https://github.com/learningequality/kolibri/releases/tag/v0.15.9-beta4) 
2. While viewing a small .pdf file (for example one from the QA Channel) start scrolling down slowly but don't scroll down to the end of the document while being at the last page.
3. Observe the header section where the resource will get marked as completed before showing the completion modal.

## Video

https://user-images.githubusercontent.com/79847249/205079305-53bc48ba-c34a-46f1-bc8c-cf7111d23c96.mp4

## Usage Details
Windows 10, Ubuntu - Chrome, Firefox

**Comment by pcenov** [comment]
@radinamatic

**Comment by AllanOXDi** [comment]
Let me take a look.

**Comment by rtibbles** [comment]
This is intended behaviour, and was the purpose of the change made in addressing https://github.com/learningequality/kolibri/issues/9348 - that we now explicitly wait for the user to be at the end of a resource before displaying the completion modal, even if their progress has been set to complete.

**Comment by pcenov** [comment]
In that case we have an interesting philosophical question - is there a difference between the intended behaviour and the expected behaviour? :) I think the intended behaviour here can be confusing to a varying degree to a learner but it's also reasonable to expect that a learner would be able to scroll all the way down so definitely not a major issue...

**Comment by rtibbles** [comment]
In discussing this with @AllanOXDi - the only thing I could think of is tightening the 'page complete' requirement for the PDF renderer to require a learner to scroll all the way to the bottom of a page for it to count as complete - but that also feels kind of finicky, because then a user might not get it marked as complete when they expect it would. I think I'd prefer it this way round than the other.

**Comment by nucleogenesis** [comment]
Not sure if I should make a new issue for this, but I just found this while looking through some PDFs and seems relevant to the issue discussed above.

If you have a PDF with one page and the whole page is visible when the page loads, then you get the completion modal before you even have a chance to read anything.

Not a desirable UX but I don't have any immediate ideas on how to best address it besides adding a minimum time/engagement thing to wait ~30 sec or so before showing the modal in any case, but I'm not sure there is a single time that would work reasonably across all content.

https://github.com/user-attachments/assets/77d7e8eb-cfb7-4095-a7f2-d7a851a969a9
